pallets/rad-claims/src/lib.rs

```diff
@@ -0,0 +1,585 @@
+        type AdminOrigin: EnsureOrigin<Self::Origin>;
+    
+        /// Expected currency of the reward claim.
+        type Currency: Currency<Self::AccountId>;
+
+        /// Associated type for Event enum
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Expected longevity of a validated unsigned extrinsic.
+        ///
+        /// This parameter is used to determine the longevity of `claim` transaction
+        type Longevity: Get<Self::BlockNumber>;
+
+        /// Minimal amount that can be claimed for a reward payout.
+        ///
+        /// This constant is set via [`parameter_types`](https://substrate.dev/docs/en/knowledgebase/runtime/macros#parameter_types)
+        /// macro when configuring a runtime.
+        #[pallet::constant]
+        type MinimalPayoutAmount: Get<u128>;
+
+        /// Constant configuration parameter to store the module identifier for the pallet.
+        ///
+        /// The module identifier may be of the form ```PalletId(*b"rd/claim")``` and set
+        /// using the [`parameter_types`](https://substrate.dev/docs/en/knowledgebase/runtime/macros#parameter_types) 
+        // macro in the [`runtime/lib.rs`] file.
+        #[pallet::constant]
+        type PalletId: Get<PalletId>;
+
+        /// A configuration for base priority of unsigned transactions.
+        ///
+        /// This is exposed so that it can be tuned for particular runtime, when
+        /// multiple pallets send unsigned transactions.
+        type UnsignedPriority: Get<TransactionPriority>;
+
+        /// Weight information for extrinsics in this pallet
+        type WeightInfo: PalletWeightInfo;
+    }
+  
+
+    // ------------------------------------------------------------------------
+    // Pallet events
+    // ------------------------------------------------------------------------
+
+    // The macro generates event metadata and derive Clone, Debug, Eq, PartialEq and Codec
+    #[pallet::event]
+    // The macro generates a function on Pallet to deposit an event
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    // Additional argument to specify the metadata to use for given type
+    #[pallet::metadata(T::AccountId = "AccountId")]
+    pub enum Event<T: Config> {
+
+        /// Event triggered after a reward claim is successfully processed
+        Claimed(T::AccountId, <T as pallet_balances::Config>::Balance),
+
+        /// Event triggered when the root hash is stored
+        RootHashStored(T::Hash),
+    }
+
+
+    // ------------------------------------------------------------------------
+    // Pallet storage items
+    // ------------------------------------------------------------------------
+
+    /// Total unclaimed rewards for an account.
+    #[pallet::storage]
+	#[pallet::getter(fn get_account_balance)]
+    pub type AccountBalances<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, T::Balance>;
```

Comment:
@vedhavyas  I think I don't understand what this represents. Does it represent the accumulated amount of claims an account has already made or the outstanding amount of claims an account can make?

Reply:
hahah yes this is claimed amount. Suggested we rename this to more apt name below